utils/set-alert-policy-required-datasources.ts

```diff
@@ -0,0 +1,126 @@
+import { passedProcessArguments } from './lib/helpers';
+import { CUSTOM_EVENT, recordNerdGraphResponse } from './newrelic/customEvent';
+import {
+  fetchPaginatedGHResults,
+  filterQuickstartConfigFiles,
+  isNotRemoved,
+} from './lib/github-api-helpers';
+import Quickstart from './lib/Quickstart';
+import Alert, { SubmitSetRequiredDataSourcesMutationResult } from './lib/Alert';
+import { chunk } from './lib/nr-graphql-helpers';
+
+type QuickstartResult = {
+  name: string;
+  dataSourceIds: string[];
+};
+
+const getQuickstartNameAndDataSources = async (
+  ghUrl?: string,
+  ghToken?: string
+): Promise<{ hasFailed: boolean; results: QuickstartResult[] }> => {
+  if (!ghToken) {
+    console.error('GITHUB_TOKEN is not defined.');
+    return { hasFailed: true, results: [] };
+  }
+
+  if (!ghUrl) {
+    console.error('Github PR URL is not defined.');
+    return { hasFailed: true, results: [] };
+  }
+
+  const files = await fetchPaginatedGHResults(ghUrl, ghToken);
+
+  const quickstartNames = filterQuickstartConfigFiles(files)
+    .filter(isNotRemoved)
+    .map(({ filename }) => {
+      const quickstart = new Quickstart(filename);
+      return {
+        name: quickstart.config.title,
+        dataSourceIds: quickstart.config.dataSourceIds ?? [],
+      };
+    });
+
+  return { hasFailed: false, results: quickstartNames };
+};
+
+const setAlertPoliciesRequiredDataSources = async (
+  ghUrl: string,
+  ghToken: string | undefined
+): Promise<boolean> => {
+  const {
+    hasFailed: hasQuickstartNamesFailed,
+    results: quickstartNamesAndDataSources,
+  } = await getQuickstartNameAndDataSources(ghUrl, ghToken);
+
+  if (hasQuickstartNamesFailed) {
+    return hasQuickstartNamesFailed;
+  }
+
+  let results: SubmitSetRequiredDataSourcesMutationResult[] = [];
+
+  for (const c of chunk(Object.entries(quickstartNamesAndDataSources), 5)) {
+    const result = await Promise.all(
+      c.map(async ([_, quickstart]) => {
+        const alertPolicy = await Alert.getAlertPolicyRequiredDataSources(
+          quickstart
+        );
+
+        if (alertPolicy.alertPolicy === null) {
+          console.error(
+            `Failed to get alert policy for quickstart ${quickstart.name}`
+          );
+
+          return { errors: alertPolicy.errors };
+        }
+
+        const { id: templateId, dataSourceIds } = alertPolicy.alertPolicy;
+
+        const result = await Alert.submitSetRequiredDataSourcesMutation(
+          templateId,
+          dataSourceIds
+        );
+
+        if (result.errors) {
+          console.error(
+            `Failed to update alert policy ${templateId} for quickstart ${quickstart.name}`
+          );
+        }
+
+        return result;
+      })
+    );
+
+    results = [...results, ...result];
+  }
+
+  const hasFailed = results.some(
```

Comment:
Yeah this makes sense, my question then is what do we do with the results on line 62 from the Promise? Do we just want to run the promise and not assign any results to it?

Reply:
yes! or you could even use a reducer like this to assign the value (https://css-tricks.com/why-using-reduce-to-sequentially-resolve-promises-works/) but honestly this just a more nice to have so if it feels like it's easier to read/ work with having the returned results and then iterating over them

Reply:
I think the reason we went with this approach is because that's what happening when we assign required data sources to dashboards. Perhaps let's keep the approach of iterating over the results to maintain consistency between the two workflows.